bin/router/src/pipeline/query_plan.rs

```diff
@@ -21,9 +21,29 @@ pub async fn plan_operation_with_cache(
         StableOverrideContext::new(&app_state.planner.supergraph, request_override_context);
 
     let filtered_operation_for_plan = &normalized_operation.operation_for_plan;
+
+    if app_state.router_config.query_planner.no_plan_cache {
+        if filtered_operation_for_plan.selection_set.is_empty()
```

Comment:
Let's share the logic of producing the plan whether or not the cache is enabled.